flink-yarn-tests/src/test/java/org/apache/flink/yarn/YARNHighAvailabilityITCase.java

```diff
@@ -18,198 +18,207 @@
-
-			new JavaTestKit(actorSystem) {{
-				for (int attempt = 0; attempt < numberKillingAttempts; attempt++) {
-					new Within(timeout) {
-						@Override
-						protected void run() {
-							try {
-								ActorGateway gateway = LeaderRetrievalUtils.retrieveLeaderGateway(
-									finalHighAvailabilityServices.getJobManagerLeaderRetriever(HighAvailabilityServices.DEFAULT_JOB_ID),
-									actorSystem,
-									timeout);
-								ActorGateway selfGateway = new AkkaActorGateway(getRef(), gateway.leaderSessionID());
-
-								gateway.tell(new TestingJobManagerMessages.NotifyWhenAtLeastNumTaskManagerAreRegistered(1), selfGateway);
-
-								expectMsgEquals(Acknowledge.get());
-
-								gateway.tell(PoisonPill.getInstance());
-							} catch (Exception e) {
-								throw new AssertionError("Could not complete test.", e);
-							}
-						}
-					};
+		final YarnClient yarnClient = getYarnClient();
+		Assert.assertNotNull(yarnClient);
+
+		Assert.assertEquals(1, yarnClient.getApplications(EnumSet.of(YarnApplicationState.RUNNING)).size());
+		final ApplicationReport report1 = yarnClient.getApplications(EnumSet.of(YarnApplicationState.RUNNING)).get(0);
+		Assert.assertEquals(1, report1.getCurrentApplicationAttemptId().getAttemptId());
+
+		final ApplicationId id = report1.getApplicationId();
+
+		waitUntilCondition(
+			() -> {
+				final File jmLog = findFile("..", (dir, name) ->
+					name.contains("jobmanager.log") && dir.getAbsolutePath().contains("_01_")
+						&& dir.getAbsolutePath().contains(LOG_DIR)
+						&& dir.getAbsolutePath().contains(FORMAT.format(id.getId())));
+				if (jmLog != null) {
+					final String jmLogText = FileUtils.readFileToString(jmLog);
+					final Matcher m = PATTERN.matcher(jmLogText);
+					// match 4 times, all vertices running
+					return m.find() && m.find() && m.find() && m.find();
 				}
+				return false;
+			}, TIMEOUT.fromNow());
 
-				new Within(timeout) {
-					@Override
-					protected void run() {
-						try {
-							ActorGateway gateway = LeaderRetrievalUtils.retrieveLeaderGateway(
-								finalHighAvailabilityServices.getJobManagerLeaderRetriever(HighAvailabilityServices.DEFAULT_JOB_ID),
-								actorSystem,
-								timeout);
-							ActorGateway selfGateway = new AkkaActorGateway(getRef(), gateway.leaderSessionID());
-
-							gateway.tell(new TestingJobManagerMessages.NotifyWhenAtLeastNumTaskManagerAreRegistered(1), selfGateway);
-
-							expectMsgEquals(Acknowledge.get());
-						} catch (Exception e) {
-							throw new AssertionError("Could not complete test.", e);
-						}
-					}
-				};
-
-			}};
-		} finally {
-			if (yarnClusterClient != null) {
-				log.info("Shutting down the Flink Yarn application.");
-				yarnClusterClient.shutDownCluster();
-				yarnClusterClient.shutdown();
-			}
+		Runtime.getRuntime().exec(new String[]{
+			"/bin/sh", "-c", "kill $(ps aux | grep -v bash | grep jobmanager | grep -v grep | FS=' \\t' awk '{print $2}')"
```

Comment:
Maybe. The intention here is to cause an external(non-flink) failure on jm, triggering YARN's restart logic instead of flink internal poststop and restart.